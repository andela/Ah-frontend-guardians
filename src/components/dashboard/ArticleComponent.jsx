import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { NavLink } from 'react-router-dom';
import default_image from '../../images/default_image.jpeg';

export const Article = props => {
  const getPlainText = data => data.replace(/<(?:.|\n)*?>/gm, '');
  const slug = props.article && props.article.slug;
  const url = 'article/' + slug;
  return (
    <NavLink to={url} id="Nav">
      <div className="article-box">
        <div>
          {' '}
          <img id="articleImage" src={default_image} width="300" height="150"/>
        </div>
        <div>
          <h3>{props.article && props.article.title}</h3>
          <h5>
            {props.article &&
              props.article.description.substring(0, 50) + '...'}
          </h5>
          <div>
            <i>
              By : {props.article && props.article.author.username}
              &nbsp;Created : &nbsp;
              {moment(props.article && props.article.created_at).fromNow()}
            </i>
          </div>
        </div>
      </div>
    </NavLink>
  );
};
Article.propTypes = {
  article: PropTypes.shape({
    author: PropTypes.object,
    body: PropTypes.string,
    title: PropTypes.string,
    created_at: PropTypes.string
  })
};
export default Article;
