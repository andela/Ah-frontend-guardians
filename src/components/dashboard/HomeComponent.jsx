import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Article from './ArticleComponent';
import { connect } from 'react-redux';
import { getDataThunk } from '../../actions/ArticleActionCreator';


export class Home extends React.Component {
  render() {
    const { articleReducer, getArticle, error } = this.props;
    const articleList = articleReducer.articles || [];
    const articles = articleList.map((article, key) => (
      <Article getArticle={getArticle} article={article} key={key} />
    ));
    return (
      <div>
        <div>
          {error ? (
            <div>
              {error.articles && error.articles.detail}
              <small>...</small>
            </div>
          ) : (
            <Fragment>
              <div id="allArticles">{articles}</div>
            </Fragment>
          )}
        </div>
      </div>
    );
  }
}
Home.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      slug: PropTypes.string,
      title: PropTypes.string,
      description: PropTypes.string,
      body: PropTypes.string,
      created_at: PropTypes.string,
      updated_at: PropTypes.string,
      author: PropTypes.object
    })
  ),
  getArticle: PropTypes.func.isRequired,
  error: PropTypes.string
};
Home.defaultProps = {
  articles: [],
  error: ''
};
export const mapStateToProps = state => {
  return state;
};
export default connect(mapStateToProps)(Home);
