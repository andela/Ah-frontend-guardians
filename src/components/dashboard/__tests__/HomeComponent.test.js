import React from 'react';
import {shallow} from 'enzyme';
import {Home}  from '../HomeComponent';
import { Article } from '../ArticleComponent';
import {mapStateToProps} from '../HomeComponent';

const getDataThunkTag = jest.fn();
const getDataThunk = jest.fn();
const handleTags = jest.fn();

const props={
    articleReducer:{
        articles:[{}]
    },
    getDataThunkTag,
    getDataThunk,
    handleTags
}
describe('Test home Component', ()=>{
    it('should render home component correctly', ()=>{
        let wrapper = shallow(<Home {...props}/>);
        wrapper.instance().handleTags('all');
        wrapper.instance().handleTags();
        expect(wrapper).toMatchSnapshot();

    });

    it('should return a state', ()=>{
        let wrapper = shallow(<Home {...props}/>);
        mapStateToProps({})
        expect(mapStateToProps({})).toEqual({})

    });

    it('should should return error on failure', ()=>{
        let wrapper = shallow(<Home {...props}/>);
        wrapper.setProps({
            error:{}
        })
        expect(wrapper.instance().props.error).toEqual({})   

    });
});
