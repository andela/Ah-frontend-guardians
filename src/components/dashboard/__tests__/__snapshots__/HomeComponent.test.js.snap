// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test home Component should render home component correctly 1`] = `ShallowWrapper {}`;
